app/services/invoices/subscription_service.rb

```diff
@@ -27,14 +27,14 @@ def call
       result.invoice = invoice
 
       fee_result = ActiveRecord::Base.transaction do
-        invoice.status = invoice_status
+        context = grace_period? ? :draft : :finalize
         fee_result = Invoices::CalculateFeesService.call(
```

Comment:
before calling this service we cannot say if it will be `closed` or `finalized` invoice, as we need the calculated number to decide
inside this service we save invoice. If invoice is in status `finalized`, we would assign number to this invoice, and updating it then to `closed` will not "delete" invoice number